tests/unit/test_bot.py

```diff
@@ -24,7 +24,9 @@ def sample_config_dict():
 @pytest.fixture
 def sample_config_dict_singular(sample_config_dict):
     config = copy.deepcopy(sample_config_dict)
-    config.update({"allowed_roles": "regular_user", "rates_urls": "www.google.com"})
+    config.update(
+        {"allowed_roles": "regular_user", "rates_urls": {"url": "www.google.com"}}
```

Comment:
Honestly not a fan of this myself; "rates_urls" implies an iterable, but we previously accepted single items as well. I think it should really just always be a list.

Reply:
The confused emoji feels ambiguous as to whether you were confused by my comment vs uncertain of how to best proceed; both of which would make that a completely reasonable response. In case the issue is the former:

I mean to say that the variable name `rates_urls` sounds like it represents multiple URLs; from the name alone, I'd expect it to be an array.

95% of the time, it actually **would** be an array, which is great. But instead, this sample config shows how we also supported that `rates_urls` variable resolving to just a string. With hindsight in mind, I think this is a confusing/inconsistent overstep of what data types we expect in that config schema. I think the best way would be to always assume it's an array, even if it's an array of one element.

```python
rates_urls1 = [ 'www.url1.com', 'www.url2.com' ]  # This is the normal case that feels most appropriate given the variable name
rates_urls2 = [ 'www.url1.com' ]  # This should be the only way the data model can represent a single url
rates_urls3 = 'www.url1.com'  # This is no longer an array, but actually the primitive type that is nominally contained in an array. 
```

We supported that third case before, but I feel we shouldn't have. It breaks the schema. If we had used type hints or a proper schema validation, we'd have typed that field as `list[str] | str`, but I'm saying it should be exclusively `list[str]` to avoid confusion.

Reply:
Meanwhile I only used the confused emoji because it was the facial expression that most closely related to your distaste in the current implementation. I didn't even read that it was the confused emoji lol

Regardless, the clarification was helpful, turns out I was more confused than I thought.

Reply:
Ambiguous comment deserves ambiguous emoji. Balance.